aten/src/ATen/test/Vec256Test.cpp

```diff
@@ -0,0 +1,1091 @@
+    class LogarithmReals : public ::testing::Test {};
+
+    template <typename T>
+    class Pow : public ::testing::Test {};
+
+    template <typename T>
+    class BitwiseFloatsAdditional : public ::testing::Test {};
+
+    template <typename T>
+    class BitwiseFloatsAdditional2 : public ::testing::Test {};
+
+    template <typename T>
+    class ComplexTests : public ::testing::Test {};
+
+    template <typename T>
+    class QuantizationTests : public ::testing::Test {};
+
+    using RealFloatTestedTypes = ::testing::Types<vfloat, vdouble>;
+    using FloatTestedTypes = ::testing::Types<vfloat, vdouble, vcomplex, vcomplexDbl>;
+    using ALLTestedTypes = ::testing::Types<vfloat, vdouble, vcomplex, vlong, vint, vshort,
+        vqint8, vquint8, vqint>;
+    using QuantTestedTypes = ::testing::Types<vqint8, vquint8, vqint>;
+    using RealFloatIntTestedTypes =
+        ::testing::Types<vfloat, vdouble, vlong, vint, vshort>;
+    using FloatIntTestedTypes =
+        ::testing::Types<vfloat, vdouble, vcomplex, vcomplexDbl, vlong, vint, vshort>;
+    using SingleFloat = ::testing::Types<vfloat>;
+    using ComplexTypes = ::testing::Types<  vcomplex, vcomplexDbl>;
+
+    TYPED_TEST_CASE(Memory, ALLTestedTypes);
+
+    TYPED_TEST_CASE(Arithmetics, FloatIntTestedTypes);
+
+    TYPED_TEST_CASE(Comparison, RealFloatIntTestedTypes);
+
+    TYPED_TEST_CASE(Bitwise, FloatIntTestedTypes);
+
+    TYPED_TEST_CASE(MinMax, RealFloatIntTestedTypes);
+
+    TYPED_TEST_CASE(Interleave, RealFloatIntTestedTypes);
+
+    TYPED_TEST_CASE(SignManipulation, FloatIntTestedTypes);
+
+    TYPED_TEST_CASE(Rounding, RealFloatTestedTypes);
+
+    TYPED_TEST_CASE(SqrtAndReciprocal, FloatTestedTypes);
+
+    TYPED_TEST_CASE(SqrtAndReciprocalReal, RealFloatTestedTypes);
+
+    TYPED_TEST_CASE(Trigonometric, RealFloatTestedTypes);
+
+    TYPED_TEST_CASE(ErrorFunctions, RealFloatTestedTypes);
+
+    TYPED_TEST_CASE(Exponents, RealFloatTestedTypes);
+
+    TYPED_TEST_CASE(Hyperbolic, RealFloatTestedTypes);
+
+    TYPED_TEST_CASE(InverseTrigonometricReal, RealFloatTestedTypes);
+
+    TYPED_TEST_CASE(InverseTrigonometric, FloatTestedTypes);
+
+    TYPED_TEST_CASE(LGamma, RealFloatTestedTypes);
+
+    TYPED_TEST_CASE(Logarithm, FloatTestedTypes);
+
+    TYPED_TEST_CASE(LogarithmReals, RealFloatTestedTypes);
+
+    TYPED_TEST_CASE(Pow, RealFloatTestedTypes);
+
+    TYPED_TEST_CASE(BitwiseFloatsAdditional, RealFloatTestedTypes);
+
+    TYPED_TEST_CASE(BitwiseFloatsAdditional2, FloatTestedTypes);
+
+    TYPED_TEST_CASE(QuantizationTests, QuantTestedTypes);
+
+    TYPED_TEST(Memory, UnAlignedLoadStore) {
+        using vec_type = TypeParam;
+        using VT = ValueType<TypeParam>;
+        constexpr size_t b_size = vec_type::size() * sizeof(VT);
+        CACHE_ALIGN unsigned char ref_storage[128 * b_size];
+        CACHE_ALIGN unsigned char storage[128 * b_size];
+        // fill with gibberish
+        for (auto& x : ref_storage) {
+            x = std::rand() % 255;
```

Comment:
Please fix or output random seed, to make tests reproducable.

Reply:
test seed was added test cases

Reply:
currently, it randomly changes for each test case. but one can set static value inside the TestSeed constructor. Or can set from the testing case.